.github/workflows/e2e.yaml

```diff
@@ -12,7 +12,7 @@ on:
 
 env:
   GO_TOOLCHAIN: "golang"
-  GO_VERSION: "1.21.6"
+  GO_VERSION: "1.22.2"
```

Comment:
Woud you want to delete this param and let's setup-go version with go.mod?

```yaml
      - uses: actions/setup-go@v5
        with:
          go-version-file: go.mod
```

Reply:
I agree, that's nicer. I'll wait for the test-infra PR to merge, then update here.

Reply:
Done.